src/dag-service.js

```diff
@@ -23,10 +25,27 @@ function DAGService (blockService) {
   // this.addRecursive
 
   // get retrieves a DAGNode, using the Block Service
-  this.get = (multihash, callback) => {
-    if (!multihash) { return callback(new Error('Invalid Key')) }
+  this.get = function (multihash, callback) {
+    const isMhash = isIPFS.multihash(multihash)
+    const isPath = isIPFS.path(multihash)
+
+    if (!isMhash && !isPath) {
+      return callback(new Error('Invalid Key'))
+    }
+
+    if (isMhash) {
+      this.getWith(multihash, callback)
+    }
+
+    if (isPath) {
+      var ipfsKey = new Buffer(base58.decode(multihash.replace('/ipfs/', '')))
+      this.getWith(ipfsKey, callback)
+    }
+  }
 
-    this.bs.getBlock(multihash, (err, block) => {
+  this.getWith = function (key, callback) {
+    const formatted = typeof key === 'string' ? new Buffer(base58.decode(key)) : key
```

Comment:
This format check is needed in this implementation. While the 'is-ipfs' module can handle any input, it only returns true or false and does not return a formatted key enforced to be a base58 decoded buffer.

'is-ipfs' could return a formatted key (buffer instead of string) if the key/path are valid multihashes, but then the name of the module wouldn't make much sense any more. And would probably need to be something like 'ipfs-format-key'